File: src/main/java/fr/uiytt/ageofempire/utils/ConfigParser.java
package fr.uiytt.ageofempire.utils;

import fr.uiytt.ageofempire.ConfigManager;
import org.bukkit.Location;

public class ConfigParser {

    /**
     * For a given string of coordinates, return a location in the Config World
     * @param locString a string either "x y z" or "x y z yaw pitch"
     * @return a location in {@link ConfigManager#getWorld()}
     */
    public static Location stringToLocation(String locString) {
        String[] string = locString.split(" ");

        try {
            Location loc = new Location(ConfigManager.getWorld(), Double.parseDouble(string[0]), Double.parseDouble(string[1]), Double.parseDouble(string[2]));
            if (string.length > 3) {
                loc.setYaw(Float.parseFloat(string[3]));
                loc.setPitch(Float.parseFloat(string[4]));
            }
            return loc;
        } catch (NumberFormatException exception) {
            exception.printStackTrace();
            return new Location(ConfigManager.getWorld(),0d,80d,0d);
        }

    }
}

File: src/main/kotlin/fr/uiytt/ageofempire/utils/ConfigParser.kt
package fr.uiytt.ageofempire.utils

import fr.uiytt.ageofempire.ConfigManager
import fr.uiytt.ageofempire.getConfigManager
import org.bukkit.Location

object ConfigParser {
    /**
     * For a given string of coordinates, return a location in the Config World
     * @param locString a string either "x y z" or "x y z yaw pitch"
     * @return a location in [ConfigManager.world]
     */
    @JvmStatic
    fun stringToLocation(locString: String): Location {
        val string = locString.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        return try {
            val loc = Location(getConfigManager().world, string[0].toDouble(), string[1].toDouble(), string[2].toDouble())
            if (string.size > 3) {
                loc.yaw = string[3].toFloat()
                loc.pitch = string[4].toFloat()
            }
            loc
        } catch (exception: NumberFormatException) {
            exception.printStackTrace()
            Location(getConfigManager().world, 0.0, 80.0, 0.0)
        }
    }
}